wagtail_live_debug/static/wagtail_live_debug/js/channels.js

```diff
@@ -0,0 +1,108 @@
+document.addEventListener('DOMContentLoaded', function() {
+    let newChannelForm = document.querySelector('#new-channel-form');
+    newChannelForm.addEventListener('submit', createChannel);
+
+    let deleteChannelBtn = document.querySelectorAll(".delete-btn");
+    deleteChannelBtn.forEach(btn => {
+        btn.addEventListener("click", deleteChannel);
+    });
+});
+
+
+async function createChannel(event) {
+    // Prevent default form submission
+    event.preventDefault();
+  
+    let form = event.target;
+    let newChannelName = form["channel_name"].value;
+    var regexp = /^[a-zA-Z0-9-_]+$/;
+    if (!newChannelName.match(regexp)) {
+        let errorMsg = "Channel names con only contain alphanumeric characters or - and _";
+        showNotif(errorMsg, reason=0);
+        // Clear out composition fields
+        document.querySelector('#channel_name').value = '';
+        return;
+    }
+
+    // Send a POST request to save channel
+    try {
+        let payload = JSON.stringify({channel_name: newChannelName});
+        let response = await fetchHelper('channels/', 'POST', payload); 
+        if (response.status != 201) {
+            // Display corresponding error message
+            let errorMsg = await response.json(); 
+            showNotif(errorMsg["channel_name"][0], reason=0);
+            // Clear out composition fields
+            document.querySelector('#channel_name').value = '';
+            return 
+        } else {
+            let channel = await response.json();
+            showNotif(`Dummy channel ${channel["channel_name"]} created.`)
+            let channelDiv = createChannelDiv(channel);
+            document.querySelector("ul").insertAdjacentElement("afterbegin", channelDiv);
+            // Clear out composition fields
+            document.querySelector('#channel_name').value = '';
+            return
+        }
+  
+    } catch(error) {
+        /* Need better error handling */
```

Comment:
Can we do that? OR Add a `TODO` so all todos can be found.

Reply:
How to add TODO in js? I don't know type of error we can expect. I will do it later.

Reply:
`/* Need better error handling */` -> `/* TODO: Need better error handling */`

Now you can search for `TODO` and find all places we need to work on.

Reply:
Added a TODO.